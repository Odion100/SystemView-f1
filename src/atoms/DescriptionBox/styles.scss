.description-box {
  width: 100%;

  &__textbox {
    min-width: 100%;
    max-width: 100%;
    min-height: 146px;
    max-height: 146px;
    background: white;
    color: black;
    font-size: 15px;
    font-family: inherit;
    padding: 8px;
    border-radius: 2px;
  }
}
