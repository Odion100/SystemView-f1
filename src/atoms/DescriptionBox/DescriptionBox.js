import React from "react";
import "./styles.scss";

const DescriptionBox = ({ text, setValue }) => {
  return (
    <div className="description-box">
      <textarea
        className="description-box__textbox"
        name="description-box"
        id="description-box"
        value={text}
        onBlur={console.log}
        onChange={
          setValue
            ? (e) => {
                setValue(e.target.value);
              }
            : null
        }
      ></textarea>
    </div>
  );
};

export default DescriptionBox;
