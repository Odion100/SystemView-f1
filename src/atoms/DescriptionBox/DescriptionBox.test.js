/* eslint-disable jest/valid-expect */
import React from "react";
import ReactDom from "react-dom";
import { render, cleanup, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import DescriptionBox from "./DescriptionBox";

afterEach(cleanup);

describe("<DescriptionBox/>", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDom.render(<DescriptionBox />, div);
  });

  it("apply text using the text prop", () => {
    const { getByRole } = render(<DescriptionBox text={"test"} />);
    expect(getByRole("textbox")).toHaveTextContent("test");
  });

  it("take text from user input", () => {
    render(<DescriptionBox />);

    screen.getByRole("textbox");
    userEvent.type(screen.getByRole("textbox"), "user input");

    expect(screen.getByRole("textbox")).toHaveValue("user input");
  });

  it("should match snapshot", () => {
    const tree = renderer.create(<DescriptionBox />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
