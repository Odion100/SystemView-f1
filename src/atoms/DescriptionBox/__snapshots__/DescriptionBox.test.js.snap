// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DescriptionBox/> should match snapshot 1`] = `
<div
  className="description-box"
>
  <textarea
    className="description-box__textbox"
    id="description-box"
    name="description-box"
    onChange={null}
  />
</div>
`;
